crates/re_space_view_spatial/src/ui_3d.rs

```diff
@@ -444,53 +508,60 @@ pub fn view_3d(
-    if response.double_clicked() {
-        // Clear out tracked camera if there's any.
-        state.state_3d.tracked_camera = None;
-        state.state_3d.camera_before_tracked_camera = None;
-
-        // While hovering an entity, focuses the camera on it.
-        if let Some((Item::InstancePath(_, instance_path), _)) = ctx.hovered().first() {
-            if let Some(camera) = find_camera(space_cameras, &instance_path.entity_path) {
-                state.state_3d.camera_before_tracked_camera =
-                    state.state_3d.orbit_eye.map(|eye| eye.to_eye());
-                state.state_3d.interpolate_to_eye(camera);
-                state.state_3d.tracked_camera = Some(instance_path.entity_path.clone());
-            } else if let Some(SelectedSpaceContext::ThreeD {
-                pos: Some(clicked_point),
-                ..
-            }) = ctx.selection_state().hovered_space_context()
-            {
-                if let Some(mut new_orbit_eye) = state.state_3d.orbit_eye {
-                    // TODO(andreas): It would be nice if we could focus on the center of the entity rather than the clicked point.
-                    //                  We can figure out the transform/translation at the hovered path but that's usually not what we'd expect either
-                    //                  (especially for entities with many instances, like a point cloud)
-                    new_orbit_eye.orbit_radius = new_orbit_eye.position().distance(*clicked_point);
-                    new_orbit_eye.orbit_center = *clicked_point;
-                    state.state_3d.interpolate_to_orbit_eye(new_orbit_eye);
+    // Track focused entity if any.
+    if let Some(focused_item) = ctx.focused_item {
+        let focused_entity = match focused_item {
+            Item::StoreId(_) | Item::DataBlueprintGroup(_, _, _) | Item::Container(_) => None,
+
+            Item::SpaceView(space_view_id) => {
+                if space_view_id == &query.space_view_id {
+                    state
+                        .state_3d
+                        .reset_camera(&state.bounding_boxes.accumulated, &view_coordinates);
                 }
+                None
+            }
+
+            Item::ComponentPath(component_path) => Some(&component_path.entity_path),
+
+            Item::InstancePath(space_view, path) => {
+                // If this is about a specific space view, focus only if it's this one.
+                // (if it's about any space view, focus regardless of which one)
+                if space_view.is_none() || space_view == &Some(query.space_view_id) {
+                    Some(&path.entity_path)
+                } else {
+                    None
+                }
+            }
+        };
+        if let Some(entity_path) = focused_entity {
+            // For the moment we only track cameras.
+            // We have everything in place to track arbitrary objects but it's a bit odd sometimes, so only focus on them instead.
```

Comment:
By "focus" you mean, move camera to them now, but don't track over time?

Reply:
yes, will clarify the comment